### PR TITLE
Only run CI builds once per PR

### PR DESCRIPTION
**Pull request by jonrohan**
I was noticing that the CI builds run twice per Pull Request.

image

This PR updates the action script to run on pushes on main and any commits to a pr.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```
Comment (manuelpuyol) [inline .github/workflows/ci.yml:7]: I'm 99% sure we had to have both because of forks

Comment (jonrohan) [reply .github/workflows/ci.yml:7]: We can use `pull_request_target` which is what I do on primer/css that will run on forks

Comment (jonrohan) [reply .github/workflows/ci.yml]: Example here https://github.com/primer/css/pull/1357

**Review by manuelpuyol — DISMISSED** [review]

**Comment by jonrohan** [comment]
> wait

uh oh

**Pull request merged**